plugin/pki/vcert.go

```diff
@@ -107,3 +130,124 @@ func (b *backend) ClientVenafi(ctx context.Context, s logical.Storage, data *fra
 	return client, role.ServerTimeout, nil
 
 }
+
+func (b *backend) getConfig(ctx context.Context, s logical.Storage, data *framework.FieldData, req *logical.Request, roleName string) (
+	*vcert.Config, error) {
+	var cfg *vcert.Config
+	b.Logger().Debug(fmt.Sprintf("Using role: %s", roleName))
+	if roleName == "" {
+		return nil, fmt.Errorf("Missing role name")
+	}
+
+	role, err := b.getRole(ctx, req.Storage, roleName)
+	if err != nil {
+		return nil, err
+	}
+	if role == nil {
+		return nil, fmt.Errorf("Unknown role %v", role)
+	}
+
+	var trustBundlePEM string
+	if role.TrustBundleFile != "" {
+		b.Logger().Debug("Reading trust bundle from file %s\n", role.TrustBundleFile)
+		trustBundle, err := ioutil.ReadFile(role.TrustBundleFile)
+		if err != nil {
+			return cfg, err
+		}
+		trustBundlePEM = string(trustBundle)
+	}
+
+	if role.Fakemode {
+		b.Logger().Debug("Using fakemode to issue certificate")
+		cfg = &vcert.Config{
+			ConnectorType: endpoint.ConnectorTypeFake,
+			LogVerbose:    true,
+		}
+
+	} else if role.TPPURL != "" && role.TPPUser != "" && role.TPPPassword != "" {
+		b.Logger().Debug("Using Venafi Platform with URL %s to issue certificate\n", role.TPPURL)
+		if trustBundlePEM != "" {
+			cfg = &vcert.Config{
+				ConnectorType:   endpoint.ConnectorTypeTPP,
+				BaseUrl:         role.TPPURL,
+				ConnectionTrust: trustBundlePEM,
+				Credentials: &endpoint.Authentication{
+					User:     role.TPPUser,
+					Password: role.TPPPassword,
+				},
+				Zone:       role.Zone,
+				LogVerbose: true,
+			}
+		} else {
+			cfg = &vcert.Config{
+				ConnectorType: endpoint.ConnectorTypeTPP,
+				BaseUrl:       role.TPPURL,
+				Credentials: &endpoint.Authentication{
+					User:     role.TPPUser,
+					Password: role.TPPPassword,
+				},
+				Zone:       role.Zone,
+				LogVerbose: true,
+			}
+		}
+
+	} else if role.URL != "" && role.AccessToken != "" {
+		if trustBundlePEM != "" {
+			cfg = &vcert.Config{
+				ConnectorType:   endpoint.ConnectorTypeTPP,
+				BaseUrl:         role.URL,
+				ConnectionTrust: trustBundlePEM,
+				Credentials: &endpoint.Authentication{
+					AccessToken:  role.AccessToken,
+					RefreshToken: role.RefreshToken,
+				},
+				Zone:       role.Zone,
+				LogVerbose: true,
+			}
+		}else {
+			cfg = &vcert.Config{
+				ConnectorType:   endpoint.ConnectorTypeTPP,
+				BaseUrl:         role.URL,
+				Credentials: &endpoint.Authentication{
+					AccessToken:  role.AccessToken,
+					RefreshToken: role.RefreshToken,
+				},
+				Zone:       role.Zone,
+				LogVerbose: true,
+			}
+		}
+	} else if role.Apikey != "" {
+		b.Logger().Debug("Using Venafi Cloud to issue certificate")
+		if trustBundlePEM != "" {
+			cfg = &vcert.Config{
+				ConnectorType:   endpoint.ConnectorTypeCloud,
+				BaseUrl:         role.CloudURL,
```

Comment:
We want this to be using the `role.URL` instead of `role.CloudURL`

Reply:
This case is being supported by @rvelaVenafi 's changes, he's making the changes to use url instead of tpp_url and cloud_url